src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -79,11 +79,13 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
   @Query(
     "SELECT count(v) > 0 FROM Visit v " +
       "WHERE v.sessionTemplateReference = :sessionTemplateReference AND " +
-      "(cast(:visitsFromDate as date)  is null or cast(v.visitStart as date) >= :visitsFromDate) ",
+      "(cast(:visitsFromDate as date)  is null or cast(v.visitStart as date) >= :visitsFromDate) AND " +
+      "((:#{#visitStatuses == null} = true) OR v.visitStatus IN (:visitStatuses))",
   )
```

Comment:
new syntax for me # does this check for an empty list?

Reply:
Yes, got it from here - https://hibernate.atlassian.net/browse/HHH-15968 but maybe as you suggested adding them to the query might make more sense.

Reply:
>add them to the query itself
I think thats the general approach we have used in VSiP.